Sources/Telegrammer/Handlers/CommandHandler.swift

```diff
@@ -65,7 +68,20 @@ public class CommandHandler: Handler {
         let types = entities.compactMap { (entity) -> String? in
             let start = text.index(text.startIndex, offsetBy: entity.offset)
             let end = text.index(start, offsetBy: entity.length-1)
-            return String(text[start...end])
+            let command = text[start...end]
+            // If the user specifies the bot using "@"
+            // and `botUsername` is not nil,
+            // check the bot name and then ignore it for further match.
+            let split = command.split(separator: "@")
+            if command.contains("@"),
```

Comment:
Use split.count == 2, it’s more efficient. In this case you can use split[0] and split [1] then. And avoid multiple optional unwrapping.

Reply:
Thank you, I've made the changes.